packages/dx-scheduler-core/src/plugins/month-view/computeds.js

```diff
@@ -15,21 +41,85 @@ export const monthCells = (currentDate, firstDayOfWeek) => {
     .year(prevMonth.year())
     .month(prevMonth.month())
     .date(prevMonthStartDay)
-    .startOf('date');
+    .startOf('day');
 
   const result = [];
-  while (result.length < WEEK_COUNT) {
+  while (result.length < (Math.trunc((MONTH_LENGTH * intervalCount) / DAY_COUNT) + 2)) {
     const week = [];
     while (week.length < DAY_COUNT) {
       week.push({
         value: from.toDate(),
-        isOtherMonth: from.month() !== currentMonth,
+        isOtherMonth: currentMonths.findIndex(month => month === from.month()) === -1,
         isCurrent: currentDay === from.date() && from.month() === currentMonth,
       });
       from.add(1, 'day');
     }
     result.push(week);
   }
-
   return result;
 };
+
+const calculateDateIntervals = (
+  appointments,
+  leftBound, rightBound,
+  monthCells,
+) => appointments
+  .map(({ start, end, ...restArgs }) =>
+    ({ start: moment(start), end: moment(end), ...restArgs }))
+  .filter(appointment =>
+    viewPredicate(appointment, leftBound, rightBound))
+  .reduce((acc, appointment) =>
+    ([...acc, ...sliceAppointmentByWeek(appointment, monthCells)]), [])
+  .filter(appointment =>
+    viewPredicate(appointment, leftBound, rightBound));
```

Comment:
What for do you apply the `viewPredicate` twice?

Reply:
It must have. The first filter by a boundary, second - by boundary after slice.